ehrql/backends/tpp.py

```diff
@@ -535,3 +535,139 @@ def medications(self):
+            FROM WL_ClockStops
+        """
+    )
+
+    wl_openpathways_raw = QueryTable(
+        # columns passed to CONVERT are likely to contain sha256 hashes
+        """
+            SELECT
+                Patient_ID AS patient_id,
+                ACTIVITY_TREATMENT_FUNCTION_CODE AS activity_treatment_function_code,
+                Current_Pathway_Period_Start_Date AS current_pathway_period_start_date,
+                PRIORITY_TYPE_CODE AS priority_type_code,
+                CONVERT(
+                    varchar(max),
+                    PSEUDO_ORGANISATION_CODE_PATIENT_PATHWAY_IDENTIFIER_ISSUER,
+                    2
+                ) AS pseudo_organisation_code_patient_pathway_identifier_issuer,
+                CONVERT(
+                    varchar(max),
+                    PSEUDO_PATIENT_PATHWAY_IDENTIFIER,
+                    2
+                ) AS pseudo_patient_pathway_identifier,
+                CONVERT(
+                    varchar(max),
+                    Pseudo_Referral_Identifier,
+                    2
+                ) AS pseudo_referral_identifier,
+                REFERRAL_REQUEST_RECEIVED_DATE AS referral_request_received_date,
+                REFERRAL_TO_TREATMENT_PERIOD_END_DATE AS referral_to_treatment_period_end_date,
+                REFERRAL_TO_TREATMENT_PERIOD_START_DATE AS referral_to_treatment_period_start_date,
+                SOURCE_OF_REFERRAL AS source_of_referral,
+                Waiting_List_Type AS waiting_list_type,
+                Week_Ending_Date AS week_ending_date
+            FROM WL_OpenPathways
+        """
+    )
+
+    wl_openpathways = QueryTable(
+        """
+            SELECT
+                Patient_ID AS patient_id,
+                ACTIVITY_TREATMENT_FUNCTION_CODE AS activity_treatment_function_code,
+                CONVERT(DATE, Current_Pathway_Period_Start_Date, 23) AS current_pathway_period_start_date,
+                CASE PRIORITY_TYPE_CODE
+                    WHEN '1' THEN 'routine'
+                    WHEN '2' THEN 'urgent'
+                    WHEN '3' THEN 'two week wait'
+                END AS priority_type_code,
+                CONVERT(
+                    varchar(max),
+                    PSEUDO_ORGANISATION_CODE_PATIENT_PATHWAY_IDENTIFIER_ISSUER,
+                    2
+                ) AS pseudo_organisation_code_patient_pathway_identifier_issuer,
+                CONVERT(
+                    varchar(max),
+                    PSEUDO_PATIENT_PATHWAY_IDENTIFIER,
+                    2
+                ) AS pseudo_patient_pathway_identifier,
+                CONVERT(
+                    varchar(max),
+                    Pseudo_Referral_Identifier,
+                    2
+                ) AS pseudo_referral_identifier,
+                CONVERT(DATE, REFERRAL_REQUEST_RECEIVED_DATE, 23) AS referral_request_received_date,
+                CONVERT(DATE, REFERRAL_TO_TREATMENT_PERIOD_END_DATE, 23) AS referral_to_treatment_period_end_date,
```

Comment:
Some of the WL table date fields contain ceiling values (see #1523); this is one of them.  Do we want to replace ceiling dates with Null? (This may be too early to do it, since it's still in the exploratory stage)

Reply:
I think that we do, as any exploratory work can be done with the `raw_` tables. I'll replace ceiling dates with NULLs.

Reply:
Thanks for your comment.